silkrpc/core/gas_price_oracle_test.cpp

```diff
@@ -0,0 +1,593 @@
+    const evmc::address &beneficiary,
+    const intx::uint256 &base_fee,
+    const intx::uint256 &max_priority_fee_per_gas_tx1,
+    const intx::uint256 &max_fee_per_gas_tx1,
+    const intx::uint256 &max_priority_fee_per_gas_tx2,
+    const intx::uint256 &max_fee_per_gas_tx2) {
+
+    silkworm::BlockWithHash block_with_hash;
+
+    block_with_hash.block.header.number = block_number;
+    block_with_hash.block.header.beneficiary = beneficiary;
+    block_with_hash.block.header.base_fee_per_gas = base_fee;
+
+    block_with_hash.block.transactions.resize(2);
+    block_with_hash.block.transactions[0].max_priority_fee_per_gas = max_priority_fee_per_gas_tx1;
+    block_with_hash.block.transactions[0].max_fee_per_gas = max_fee_per_gas_tx1;
+    block_with_hash.block.transactions[0].from = kFromTnx1;
+
+    block_with_hash.block.transactions[1].max_priority_fee_per_gas = max_priority_fee_per_gas_tx2;
+    block_with_hash.block.transactions[1].max_fee_per_gas = max_fee_per_gas_tx2;
+    block_with_hash.block.transactions[1].from = kFromTnx2;
+
+    return block_with_hash;
+}
+
+static void clear_blocks_vector() {
+    kBlocks.clear();
+}
+
+static void fill_blocks_vector(uint16_t size,
+    const evmc::address &beneficiary,
+    const intx::uint256 &base_fee,
+    const intx::uint256 &max_priority_fee_per_gas_tx1,
+    const intx::uint256 &max_fee_per_gas_tx1,
+    const intx::uint256 &max_priority_fee_per_gas_tx2,
+    const intx::uint256 &max_fee_per_gas_tx2) {
+
+    kBlocks.reserve(size);
+    for (auto idx = 0; idx < size; idx++) {
+        silkworm::BlockWithHash block_with_hash = allocate_block(idx, beneficiary, base_fee,
+            max_priority_fee_per_gas_tx1, max_fee_per_gas_tx1,
+            max_priority_fee_per_gas_tx2, max_fee_per_gas_tx2);
+        kBlocks.push_back(block_with_hash);
+    }
+}
+
+static void fill_blocks_vector(uint16_t size,
+    const evmc::address &beneficiary,
+    const intx::uint256 &base_fee,
+    const intx::uint256 &max_priority_fee_per_gas,
+    const int delta_max_priority_fee_per_gas,
+    const intx::uint256 &max_fee_per_gas,
+    const int delta_max_fee_per_gas) {
+
+    kBlocks.reserve(size);
+    for (auto idx = 0; idx < size; idx++) {
+        int64_t max_priority = int64_t(max_priority_fee_per_gas) + delta_max_priority_fee_per_gas * idx;
+        if (max_priority < 0) {
+            max_priority = 0;
+        }
+        int64_t  max_fee = int64_t(max_fee_per_gas) + delta_max_fee_per_gas * idx;
+        if (max_fee < 0) {
+            max_fee = 0;
+        }
+
+        silkworm::BlockWithHash block_with_hash = allocate_block(idx, beneficiary, base_fee,
+            intx::uint256{max_priority},
+            intx::uint256{max_fee},
+            intx::uint256{max_priority},
+            intx::uint256{max_fee});
+        kBlocks.push_back(block_with_hash);
+    }
+}
+
+asio::awaitable<silkworm::BlockWithHash> get_block(uint64_t block_number) {
```

Comment:
This function will probably be unuseful after suggested changes

Reply:
It is not possible to remove due to compiler error using lambda as coroutine.